### PR TITLE
Add spotless and checkstyle

### PR DESCRIPTION
**Pull request by eddumelendez**

### CODE REVIEW

**Review by bsideup — APPROVED** [review]
OMG this is going to be awesome 🔥 🚀

File: gradle/ci-support.gradle [hunk 1]
```diff
@@ -29,6 +29,11 @@ gradle.taskGraph.whenReady {
             subproject.tasks.findByName("javadoc")?.enabled = false
             subproject.tasks.findByName("delombok")?.enabled = false
             subproject.tasks.findByName("japicmp")?.enabled = false
+            subproject.tasks.findByName("spotlessJavaCheck")?.enabled = false
+            subproject.tasks.findByName("spotlessGroovyGradleCheck")?.enabled = false
+            subproject.tasks.findByName("checkstyleMain")?.enabled = false
+            subproject.tasks.findByName("checkstyleTest")?.enabled = false
+            subproject.tasks.findByName("checkstyleJarFileTest")?.enabled = false
```
Comment (bsideup) [inline gradle/ci-support.gradle:36]: why do we need to disable them here? :)

Comment (eddumelendez) [reply gradle/ci-support.gradle:36]: I need to disable the analysis over `jarFileTest` folder. But trying a global config and delete this one

Comment (bsideup) [reply gradle/ci-support.gradle:36]: Perhaps we could do something like `tasks.ofType(SpotlessCheck.class)` (artificial name) to disable them all in the root build file without listing?

File: build.gradle [hunk 2]
```diff
@@ -105,4 +108,39 @@ subprojects {
...
+        configFile = rootProject.file('config/checkstyle/checkstyle.xml')
+        checkstyleTest.enabled = false
+        checkstyleMain.enabled = false
+    }
+
+    tasks.whenTaskAdded {task ->
+        if (task.name.equals("checkstyleJarFileTest")) {
+            task.enabled = false
+        }
+    }
```
Comment (bsideup) [inline build.gradle:145]: I think you can try this:
```groovy
tasks.withType(Checkstyle.class)*.enabled = false
```

You _may_ need to wrap it with `project.afterEvaluate` tho, but probably not

Comment (bsideup) [reply build.gradle:145]: Another option would be to set `ignoreFailures = true` on `checkstyle`, btw

File: build.gradle [hunk 3]
```diff
@@ -105,4 +108,34 @@ subprojects {
...
+            target '**/*.groovy'
+            greclipse('4.19.0')
+        }
+    }
+
+    checkstyle {
+        toolVersion = "9.3"
+        configFile = rootProject.file('config/checkstyle/checkstyle.xml')
+        checkstyleTest.enabled = false
+        checkstyleMain.enabled = false
```
Comment (bsideup) [inline build.gradle:138]: We probably don't need these now that the failures are ignored?

**Pull request merged**